From a7587b78b4c2c46e2ec9612903b10b75e3adf973 Mon Sep 17 00:00:00 2001
From: Viktor Ashirov <vashirov@redhat.com>
Date: Sat, 8 Nov 2025 00:44:35 +0100
Subject: [PATCH] ipa-lockout: fix memory leaks

Move cleanup of `unlock_time` to `done` label to ensure cleanup in all code paths.

Related: https://pagure.io/freeipa/issue/9895
Signed-off-by: Viktor Ashirov <vashirov@redhat.com>
Reviewed-By: Rob Crittenden <rcritten@redhat.com>
Reviewed-By: Alexander Bokovoy <abbra@users.noreply.github.com>
---
 daemons/ipa-slapi-plugins/ipa-lockout/ipa_lockout.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/daemons/ipa-slapi-plugins/ipa-lockout/ipa_lockout.c b/daemons/ipa-slapi-plugins/ipa-lockout/ipa_lockout.c
index ecb6d6df52b7e3c2239884e173c8498e651dff34..5142c79bcfcfa985a35d45f689a7c1d1ad3f4feb 100644
--- a/daemons/ipa-slapi-plugins/ipa-lockout/ipa_lockout.c
+++ b/daemons/ipa-slapi-plugins/ipa-lockout/ipa_lockout.c
@@ -813,7 +813,6 @@ static int ipalockout_preop(Slapi_PBlock *pb)
                 goto done;
             }
         }
-        slapi_ch_free_string(&unlock_time);
     }
 
     max_fail = slapi_entry_attr_get_uint(policy_entry, "krbPwdMaxFailure");
@@ -840,6 +839,7 @@ static int ipalockout_preop(Slapi_PBlock *pb)
 
 done:
     if (lastfail) slapi_ch_free_string(&lastfail);
+    if (unlock_time) slapi_ch_free_string(&unlock_time);
     slapi_entry_free(target_entry);
     slapi_entry_free(policy_entry);
     if (values != NULL) {
-- 
2.52.0

